### DISCUSSION TITLE
Can't connect BitcrazeVM !!

### DISCUSSION THREAD

**Discussion by Reza1973** [bitcraze/discussions#688] [General]
I am trying to connect by VM but I am receiving this error:
![error](https://user-images.githubusercontent.com/131666379/233973770-bea8c275-090e-4e36-8b14-98fabb44c487.jpg)

I have tried to use terminal but it is same and the error code is: 

Result Code: | NS_ERROR_FAILURE (0X80004005)
-- | --

Any idea, what I need to do?

**Comment by krichardsson** [comment]
No idea what this might be. Can you run any VM images at all? Looks more like a Virtual Box problem to me.

**Reply by Reza1973** [reply]
Thanks Kristoffer, Yes see the attached screenshot, 
<img width="1800" alt="Screenshot 2023-04-24 at 13 03 33" src="https://user-images.githubusercontent.com/131666379/233991391-4a1948d9-eb63-480a-b3d3-32063f81b68a.png">
you see the other one up and running

**Reply by ataffanel** [reply]
Looking around the internet for the error you are getting I find things about the Linux virtual box module (not applying to you there ...) and things like USB support and the expansion pack.

The VM is configured by default to use USB3 that might require the expansion pack. You can download and install it from the [virtualbox website](https://www.virtualbox.org/wiki/Downloads). You could also edit the VM configuration to switch the USB controller to USB2.

By the way, if that does not work, what kind of Mac do you have? X86 or a newer M1/M2 ARM based Mac?

**Reply by CalvinKoung** [reply]
I encounter the same problem. Is there a solution ?

**Reply by ataffanel** [reply]
If you are running an apple-silicon based mac (M1 or M2), no, as far as I know there is no solution to run the VM, you need to run the Crazyflie tools natively (see bellow for my other answer)

**Reply by CalvinKoung** [reply]
OK, sure. thanks

**Comment by Reza1973** [comment]
It is MacBook M2 pro.
I reinstalled the M2 compatible version which is available in Virtual box website (it is "Developer preview for macOS / Arm64 (M1/M2) hosts") and run the CrazyflieVM again.
It is trying to run but VM is aborting after a few second.
I have attached a short video of what is happening. 

https://user-images.githubusercontent.com/131666379/234544237-dca13130-ef09-4875-a1c9-af3f8c5497ea.mov

**Reply by ataffanel** [reply]
I can observe the same things on our Mac M1. Virtualbox for Apple silicon is still in developer preview and is apparently not capable of running the Bitcraze VM yet. I tried with VMWare fusion but it was not able to run an x86 VM. So I am afraid that running the VM on apple silicon is not yet possible. We will update the documentation to reflect that.

I suggest you install the tools directly on your mac (our documentation is updated for M1 macs) or on a manually installed Linux VM. I am assuming you will have much better performance installing ARM64-version of Linux in VMs though.
